### PR TITLE
Move test files to subfolders and update pytest's conftest.py

### PR DESCRIPTION
**Pull request by CunliangGeng**
To solve issue #99.

### CODE REVIEW

File: tests/pairedomics/test_downloader.py [hunk 1]
```diff
@@ -101,4 +101,3 @@ def test_extract_metabolomics_data(tmp_path):
     assert (Path(sut.project_file_cache) / "networkedges_selfloop/6da5be36f5b14e878860167fa07004d6.pairsinfo").is_file()
```
Comment (hechth) [inline tests/pairedomics/test_downloader.py:101]: Why remove this?

File: tests/scoring/test_data_links.py [hunk 2]
```diff
@@ -1,7 +1,14 @@
 from nplinker.scoring.linking.data_linking import DataLinks
-from .test_metabolomics import spec_dict, spec_with_families
-from .test_gnps_molecular_family_loader import molecular_families_gnps
+from nplinker.metabolomics.gnps.gnps_molecular_family_loader import GNPSMolecularFamilyLoader
+from .. import DATA_DIR
+import os
+import pytest
 
+@pytest.fixture
```
Comment (hechth) [inline tests/scoring/test_data_links.py:16]: Maybe move this also to conftest?

**Review by hechth — COMMENTED** [review]
See comment, otherwise looks good to me.

File: tests/pairedomics/test_downloader.py (hunk 1 above)
Comment (CunliangGeng) [reply tests/pairedomics/test_downloader.py:101]: remove what? nothing was removed here

File: tests/scoring/test_data_links.py (hunk 2 above)
Comment (CunliangGeng) [reply tests/scoring/test_data_links.py:16]: it's only used by this file, so better to keep it here. `conftest.py` is a place for fixtures shared by different modules/files.

**Comment by CunliangGeng** [comment]
@hechth Two more commits to move non-shared fixtures from `conftest.py` to the specific file that uses them. Ready to review.

**Comment by CunliangGeng** [comment]
We merged the newer PR first so leading to conflicts. If there is no other comments for this PR, I'll solve the conflicts and merge it.

**Review by hechth — APPROVED** [review]

**Pull request merged**